device/api/umd/device/architecture_implementation.h

```diff
@@ -25,7 +25,7 @@ class architecture_implementation {
 public:
     virtual ~architecture_implementation() = default;
 
-    virtual tt::ARCH get_architecture() const = 0;
+    virtual tt::umd::Arch get_architecture() const = 0;
```

Comment:
Not really sure about this one, I think Arch may be something that could potentially go into common repo, when there is something like that. Not sure whether we should put it in umd namespace. I guess this is one of the things that are, let's say, layer below UMD so basic types that we are building umd on top of. Maybe I would like to keep this in tt namespace so we can more easily distinguish things that could go to common repo.

Reply:
Makes sense, I'll revert to tt::.
I somewhat agree on the common argument, but I wouldn't like to fuel the fire there, and would like to keep it here in the umd folder outside common. When we consider adding common utils repo, we can revisit this.

Reply:
Makes sene. I think we should still put as many things into umd namespace, but yeah seems like we are on the same page here